### PR TITLE
Box selection

### PR DESCRIPTION
**Pull request by feldtsen**
You can now drag-select the active player's selectable GameObjects. 

Note: There is one annoyance for a future update. You still pick up 1 boid by clicking a boid resulting in unwanted behaviour. Picking up should be associated with a modifier key or some kind of gesture and moving around should be available after making the wanted selection.

### CODE REVIEW

File: Assets/Prefabs/Units/Melee.prefab [hunk 1]
```diff
@@ -76,16 +77,17 @@ MonoBehaviour:
...
+  damage: 10
+  maxSpeed: 3
+  targetHeight: 2
+  collisionAvoidanceDistance: 3
+  avoidCollisionWeight: 5
+  hover_Ki: 2
+  hover_Kp: 10
+  timeBetweenAttacks: 1
+  selectionGraphic: {fileID: 1473401980719432369, guid: d1755760120f61e43879ab4b335b27b3, type: 3}
   dead: 0
```
Comment (palmdrop) [inline Assets/Prefabs/Units/Melee.prefab:91]: Why change the values of the prefab fields? I think they were intentionally set to 0 in a previous commit. The values are now edited in the constructor instead.

File: Assets/Scripts/Selection/Selectable.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using UnityEngine;
+
+public abstract class Selectable : MonoBehaviour, ISelectable
+{
+    
+    // Adds the selected visualisation 
+    public void SetSelectionIndicator(bool isSelected)
+    {
+        // TODO: now highlight indicator needs to be the second child of the Boid container, should be solved differently
+        transform.GetChild(1).gameObject.SetActive(isSelected);
```
Comment (palmdrop) [inline Assets/Scripts/Selection/Selectable.cs:10]: Maybe find by name in constructor, and then save the result in a field for later use?

**Review by palmdrop — COMMENTED** [review]
Had a few thoughts, but looks good.

File: Assets/Prefabs/Units/Melee.prefab (hunk 1 above)
Comment (feldtsen) [reply Assets/Prefabs/Units/Melee.prefab:91]: Hm, right. I will change it back.

File: Assets/Scripts/Selection/Selectable.cs (hunk 2 above)
Comment (feldtsen) [reply Assets/Scripts/Selection/Selectable.cs:10]: Sure, I can change it now if you want.

File: Assets/Prefabs/Units/Melee.prefab (hunk 1 above)
Comment (feldtsen) [reply Assets/Prefabs/Units/Melee.prefab]: I actually don't know why this shows up. I tried to go to the file and the variables doe not seem to be there. I guess everything is as it should.

File: Assets/Prefabs/Units/Melee.prefab [hunk 3]
```diff
@@ -77,22 +76,7 @@ MonoBehaviour:
...
-  avoidCollisionWeight: 5
-  hover_Ki: 2
-  hover_Kp: 10
-  timeBetweenAttacks: 1
-  selectionGraphic: {fileID: 1473401980719432369, guid: d1755760120f61e43879ab4b335b27b3, type: 3}
-  dead: 0
-  mesh: {fileID: 0}
-  collisionMask:
-    serializedVersion: 2
-    m_Bits: 0
```
Comment (feldtsen) [inline Assets/Prefabs/Units/Melee.prefab]: So it seems to have fixed itself.

**Review by palmdrop — APPROVED** [review]
Well done, works perfectly. Although it's yet not possible to remove/move selected boids? Approved though.

**Comment by feldtsen** [comment]
> Well done, works perfectly. Although it's yet not possible to remove/move selected boids? Approved though.

That is correct. This is just the implementation for box selection.

**Pull request merged**